packages/editor/src/components/post-publish-panel/maybe-tags-panel.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { Dashicon, PanelBody } from '@wordpress/components';
+
+import FlatTermSelector from '../post-taxonomies/flat-term-selector';
+
+const TagsPanel = () => <PanelBody initialOpen={ true } title={ [
+	<Dashicon key={ 'dashicon-lightbulb' } icon={ 'lightbulb' } />,
+	__( 'Tip:' ),
+	<span className="editor-post-publish-panel__link" key="label">{
+		__( 'Add tags to your post' )
+	}</span>,
+] }>
+	<FlatTermSelector slug={ 'post_tag' } />
+</PanelBody>;
+
+class MaybeTagsPanel extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			hadTags: props.hasTags,
```

Comment:
Can you help me understand why it is not possible to set state only once on initial load and keep it this way for the rest of interaction with this panel? Checks like:
```js
if ( currentHadTags === undefined && hasTags !== undefined ) {
```
indicate that it can be `undefined` for some reasons. If that is the reason, can we postpone mounting this component until this data is fetched?

Reply:
That's explained it in the long `getDerivedStateFromProps` comment. TLDR is: tags may be not fetched yet at this point. How to test this behavior:

- Create a new post.
- Make sure the sidebar document hasn't been opened. If it is, collapse it, and reload the page.
- Click the publish button. At this point, the pre-publish panel will be opened but the taxonomies aren't fetched yet (because the sidebar document is closed).

I'd love to get rid of this complexity, but I'm a bit fuzzy about Gutenberg fetching mechanisms. Any suggestion about code examples I should look at for inspiration and learn-up?

Reply:
Found a way to do this by using the `ifCondition` HOC component suggested by Riad, see e7dde0f.